crates/interpreter/src/opcode.rs

```diff
@@ -276,43 +284,132 @@ impl OpCode {
 }
 
 /// Information about opcode, such as name, and stack inputs and outputs.
-#[derive(Clone, Copy, Default, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct OpCodeInfo {
-    pub name: &'static str,
-    pub inputs: u8,
-    pub outputs: u8,
-    // TODO make this a bitfield
-    pub is_eof: bool,
-    // If the opcode is return from execution. aka STOP,RETURN, ..
-    pub is_terminating_opcode: bool,
-    /// Size of opcode with its intermediate bytes.
-    ///
-    /// RJUMPV is special case where the bytes len is depending on bytecode value,
-    /// for RJUMV size will be set to one byte while minimum is two.
-    pub immediate_size: u8,
+    // TODO: make this a bitfield. This is OK for now since we have a few alignment bytes to spare.
+    // Invariant: `(name_ptr, name_len)` is a `&'static str`.
+    name_ptr: NonNull<u8>,
```

Comment:
what is the benefit of this? Isn't `&'static str` exactly the same as `(ptr,len)`?

Reply:
separated to make the structure smaller since length fits in u8

Reply:
That makes sense